src/middlewares/category.middleware.js

```diff
@@ -0,0 +1,18 @@
+import {Categories} from '../database/models/index';
+export const checkIfCategoryExists = async (req, res, next) => {
+  const {name}=req.body
+    const categories = await Categories.findOne({
+      where: { name: name.toLowerCase() },
+    });
+    if (categories) return res.status(400).json({ message: 'Category already exists!' });
+    next();
+  };
+
+  export const checkCategory = async (req, res, next) => {
+    const {category} = req.body
+    const categories = await Categories.findOne({ where: { name: category.toLowerCase()} });
+    if (!categories)
+      return res.status(404).json({ message: "Product category doesn't exist" });
+    req.body.categoryId = categories.id;
+    next();
+  };
```

Comment:
- I think this middleware will be used to check if `categoryId` exists, because when adding a new product `req.body` will include `categoryId` not the category name
- Then, instead of finding a category by name, you can find it by primary key (findByPk)
- Rename middleware to `checkIfCategoryExistById`